api/http/handler/endpoints/endpoint_list.go

```diff
@@ -231,6 +234,17 @@ func filterEndpointsByTypes(endpoints []portainer.Endpoint, endpointTypes []int)
 	return filteredEndpoints
 }
 
+func filterEndpointsByEdgeDeviceFilter(endpoints []portainer.Endpoint, edgeDeviceFilter bool) []portainer.Endpoint {
```

Comment:
Minor thing but you have the word "filter" in the function name twice.    Wouldn't `filterEndpointsByEdgeDevice` be cleaner?